src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,41 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance/party or player
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+
+    uint8 id = 0;
+
+    auto PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
+
+    if (PChar != nullptr)
+    {
+        if (PChar->PParty && PChar->PParty->m_PAlliance)
+        {
+            id = PChar->PParty->m_PAlliance->m_AllianceID;
+        }
+        else if (PChar->PParty)
+        {
+            id = PChar->PParty->GetPartyID();
+        }
+        else
+        {
+            id = PChar->id;
+        }
+    }
+
+    lua_pushnumber(L, id);
```

Comment:
I feel this should push nil if you're not in an alliance. Falling back to party ID would not be the behavior I would expect on the script side.

Reply:
pushing nil now